### ISSUE TITLE
[$500] Page goes to home screen if press chat and refresh the page - reported by @phivh

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#8126]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Launch the URL https://staging.new.expensify.com/
2. Log in with any account.
3. Open any chat 
4. Refresh the page

## Expected Result:
After page refresh user should stay on the chat conversation with recent user

## Solution hints:
So if we break down the problem into steps considering https://github.com/Expensify/App/issues/8126#issuecomment-1103000322.

Manage the drawer state somehow https://github.com/Expensify/App/issues/8126#issuecomment-1103000322.
Able to navigate the state of the drawer via URL ?? (look at [linkingConfig](https://reactnavigation.org/docs/navigation-container/#linkingconfig) if you can find something).
we can think of A hybrid solution using react-navigation and Onyx.

## Actual Result:
User redirected back to LHN 

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Mobile Web

**Version Number:** 1.1.42-2
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 
Linked to https://github.com/Expensify/App/issues/5027 

https://user-images.githubusercontent.com/43995119/158106720-73d3cf1c-f2ad-4ddf-bcb7-047206d94cf6.mp4

**Expensify/Expensify Issue URL:**  
**Issue reported by:** @phivh
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1646153934144159

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @danieldoglas (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @laurenreidexpensify (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by danieldoglas** [comment]
Seems like something related to the router.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @NicMendonca (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by laurenreidexpensify** [comment]
@NicMendonca I'm out for a couple days, wanted to make sure someone picks this up before then - just saw this was I was logging off for day 😅

**Comment by NicMendonca** [comment]
Upwork job: https://www.upwork.com/jobs/~01c09e65fbbdd4bb12

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @parasharrajat (`Exported`)

**Comment by mdneyazahmad** [comment]
I reported this issue earlier but it did not get any response then https://expensify.slack.com/archives/C01GTK53T8Q/p1643463262081129

**Comment by parasharrajat** [comment]
Sorry but based on the Slack thread it was reported before your report.

**Comment by mollfpr** [comment]
## Proposal

Make the drawer `defaultStatus` to `closed` on mWeb will prevent LHN to show. To make sure that this only applied to mWeb I add some condition below.
```diff
+    if (isSmallScreenWidth && Platform.OS === 'web') {
+        return 'closed';
+    }
```
https://github.com/Expensify/App/blob/7919e24c808dea9dedfe4a2d513365cf092cc1e1/src/libs/Navigation/Navigation.js#L81-L86

## Result

https://user-images.githubusercontent.com/25520267/158653896-ed899368-1c60-4eb4-8a90-2eb4cbaf2948.MP4

**Comment by parasharrajat** [comment]
Thanks for the proposal @mollfpr. Could you please tell me what the would be the output of? LHN or report screen?
1. Open the App via localhost:8080 URL. 
2. Do 1 and refresh. 
3. Do 1 and click on any report, refresh the page. 
4. Open the app via localhost:8080/r/anyreportid  
5. Do 4 and refresh the app. 
6. Do 4 and use the back arrow to go back to LHN and refresh the app.

**Comment by mollfpr** [comment]
@parasharrajat 

1. It's open the recent report screen
2. It's staying on the report screen
3. Number 1 shows the recent report screen, so I press the back button and select another report list. After refresh is shown the recent report screen.
4. It's open the report screen from the attached reportID in the route
5. Still stay the recent report screen after refresh
6. Show the latest report screen shown

**Comment by parasharrajat** [comment]
Could you please use `LHN` or `report screen` words in your prev reply? It is hard to understand what are you referring to.

**Comment by mollfpr** [comment]
@parasharrajat updated ^

**Comment by parasharrajat** [comment]
Ok. Thanks. 
The output should be: 
1. LHN.
2. LHN.
3. Report screen.
4. Report screen.
5. Report screen.
6. LHN.

**Comment by mollfpr** [comment]
If the route is ‘REPORT_WITH_ID’ it should show the ‘report screen’ right? Even after you enter the base url with logged in account will redirect you to route ‘REPORT_WITH_ID’. 🤔

**Comment by parasharrajat** [comment]
Yup. That is an issue. Currently, navigation is configured in a way that it always loads the report screen. we need a way to keep the user on LHN when user was on LHN before refreshing and vice-versa.

**Comment by mollfpr** [comment]
Any background story why the app redirect us to ‘REPORT_WITH_ID’ route in the first place?

**Comment by parasharrajat** [comment]
WE load Drawer on Home route `/r` which will always load the first screen which is report screen `/r/id`. DefaultStatus controls whether we want to see closed(report screen) or opened(LHN). As you see the situation here is different we need to dynamically control this status.

**Comment by mdneyazahmad** [comment]
> Sorry but based on the Slack thread it was reported before your report.

@parasharrajat If I am not wrong I reported this on slack January 29th. And this one is on March 1st.

**Comment by parasharrajat** [comment]
The original report was JAN 21. The attached slack thread reshares that bug report. let me know if I am missing something.
<img width="150" src="https://user-images.githubusercontent.com/24370807/158933150-85a78f85-5a0e-4c1b-bec6-8ef2927b1391.png" />

**Comment by phivh** [comment]
@mdneyazahmad Sorry bro, this bug has reported at Jan 21st. Please take. ;)
![image](https://user-images.githubusercontent.com/5449119/158934482-b9e4cf76-1d23-4a2e-974f-3b5e919521ea.png)

**Comment by mdneyazahmad** [comment]
@phivh I am sorry. I missed the reference part. It is indeed JAN 21st.

**Comment by thienlnam** [comment]
@NicMendonca Could we please get this issue doubled? We are still awaiting and evaluating proposals

**Comment by NicMendonca** [comment]
doubled price: https://www.upwork.com/jobs/~01c09e65fbbdd4bb12

**Comment by NicMendonca** [comment]
doubled: https://www.upwork.com/jobs/~01c09e65fbbdd4bb12

**Comment by NicMendonca** [comment]
doubled: https://www.upwork.com/jobs/~01c09e65fbbdd4bb12

**Comment by Salehmrcp** [comment]
Just it need to change the code 
There are 2- one for refresh and other for chat

**Comment by mollfpr** [comment]
### Proposal

Adding a new Onyx key `IS_DRAWER_CLOSED_LAST_TIME` to save the state of the drawer should be `open` or `closed`.

On `src/libs/Navigation/Navigation.js` watch the Onyx key value
```js
let isDrawerClosedLastTime = false;
Onyx.connect({
    key: ONYXKEYS.IS_DRAWER_CLOSED_LAST_TIME,
    callback: val => isDrawerClosedLastTime = val,
});
```

Set `ONYXKEYS.IS_DRAWER_CLOSED_LAST_TIME` to `false` when the drawer is gonna be opened.
https://github.com/Expensify/App/blob/7919e24c808dea9dedfe4a2d513365cf092cc1e1/src/libs/Navigation/Navigation.js#L56-L63

Set `ONYXKEYS.IS_DRAWER_CLOSED_LAST_TIME` to `true` when navigating to the report route.
https://github.com/Expensify/App/blob/7919e24c808dea9dedfe4a2d513365cf092cc1e1/src/libs/Navigation/Navigation.js#L143-L146


The last thing to do is determined should the drawer state opened or closed on mWeb.
```js
    if (isSmallScreenWidth && Platform.OS === 'web') {
        return isDrawerClosedLastTime ? 'closed' : 'open';
    }
```
https://github.com/Expensify/App/blob/7919e24c808dea9dedfe4a2d513365cf092cc1e1/src/libs/Navigation/Navigation.js#L81
---

**Result**
1. Open the App via localhost:8080 URL.

https://user-images.githubusercontent.com/25520267/164077698-56b9e1e3-4e21-4dad-a453-450f99351c0b.mov

2. Do 1 and refresh.

https://user-images.githubusercontent.com/25520267/164077791-f977f0cc-2019-4489-96a3-1fee6fa0160d.mov


3. Do 1 and click on any report, refresh the page.

https://user-images.githubusercontent.com/25520267/164077881-70f83e35-f0e1-47d5-9e33-e243e24d7574.mov


4. Open the app via localhost:8080/r/anyreportid

https://user-images.githubusercontent.com/25520267/164078297-410da91b-c78e-4b49-84a2-4b0dd823c69c.mov


5. Do 4 and refresh the app.

https://user-images.githubusercontent.com/25520267/164078504-033131b0-12fd-4c08-9a5b-9c08b0a6ac3e.mov



6. Do 4 and use the back arrow to go back to LHN and refresh the app.

https://user-images.githubusercontent.com/25520267/164078515-b6f983e4-742b-4a89-bfb0-521171ca986b.mov

**Comment by parasharrajat** [comment]
Cases that you presented should be working already. What we need is that
- When user is on LHN and refreshes the page, it should open LHN.
- when user is on report page, and refreshed the app, it should open the reportscreen.

**Comment by mollfpr** [comment]
> Cases that you presented should be working already. What we need is that
> 
> * When user is on LHN and refreshes the page, it should open LHN.
> * when user is on report page, and refreshed the app, it should open the reportscreen.

Is the above case the same as https://github.com/Expensify/App/issues/8126#issuecomment-1103000322 result number 2 and 5?

Btw when a user is currently on LHN and opening the app with `localhost:8080/r/reportid` what is the expected result here? Should stay on LHN or report screen?

And when a user is currently on the report screen and they open the app with `localhost:8080` should it open LHN or report screen?

**Comment by parasharrajat** [comment]
let's forget the opening of the App for a moment. First, [these](https://github.com/Expensify/App/issues/8126#issuecomment-1104168082) should work. So user should land on the same page when he **refreshes* the page whatever url is. 

Using `localhost:8080` should open the Home page which is LHN.

**Comment by mollfpr** [comment]
I just check the URL path from the home screen which `/` is always navigated to `/` and then `/r/reportid`, whether we refresh the page from `localhost:8080` or `localhost:8080/r/reportid`.

Also if we refresh from `localhost:8080/r/reportid` it will navigate to `/` then `r/reportid`.

**Comment by mollfpr** [comment]
Yeah probably the limitation from the drawer, since other screens that define in the navigator is doesn't have that behavior.

**Comment by thienlnam** [comment]
Looking through the thread it seems like we're limited by the ability to control the page behavior using the URL. It looks like we're limited by the react-navigation library? What's suggested outcome here @parasharrajat, it doesn't seem like we'll be able to do this with react-navigation

**Comment by parasharrajat** [comment]
I haven't seen any proposal attempt using React-navigation to the solve the problem so I can't say the same. I think it would be actually possible to do with react-navigation. 

So if we break down the problem into steps considering https://github.com/Expensify/App/issues/8126#issuecomment-1103000322. 

1. Manage the drawer state somehow https://github.com/Expensify/App/issues/8126#issuecomment-1103000322. 
2. Able to navigate the state of the drawer via URL ?? 

we can think of A hybrid solution using react-navigation and Onyx.

**Comment by thienlnam** [comment]
Cool cool, going to add that to the description and have the job doubled

**Comment by NicMendonca** [comment]
doubled price: https://www.upwork.com/jobs/~01c09e65fbbdd4bb12

**Comment by Salehmrcp** [comment]
I am working on solving it

**Comment by Salehmrcp** [comment]
Just for confirmation please
The issue for the refresh problem
Right?

**Comment by parasharrajat** [comment]
Does this help https://github.com/Expensify/App/issues/8126#issuecomment-1104168082? Please go through the discussion on the issue to get more details.

**Comment by mollfpr** [comment]
### Proposal

1. Create Onyx key to hold value for default drawer status
https://github.com/Expensify/App/blob/a0a76d83945059c15e86e700fd74eb58500d7c59/src/ONYXKEYS.js#L195
```diff
+    DEFAULT_DRAWER_STATUS: 'defaultDrawerStatus',
```
2. Create action to set the `DEFAULT_DRAWER_STATUS` Onyx key
```js
function setDefaultDrawerStatus(defaultDrawerStatus) {
    Onyx.set(ONYXKEYS.DEFAULT_DRAWER_STATUS, defaultDrawerStatus);
}
```
3. Use `screenListeners` param on `Drawer.Navigator` to get the state every time the screen drawer change
https://github.com/Expensify/App/blob/dee673570654cd14aae5c36230aeda69bc04db13/src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js#L54-L68
```js
                screenListeners={{
                    state: (e) => {
                        // Get the current navigation screen state
                        const state = e.data.state;

                        // Get the history. If the current state has history that type to drawer,
                        // that`s mean the current route is report screen.
                        // If there`s no history then current route is home screen.
                        // But those condition only apply when navigate, not when refresh the page
                        const hasDrawerHistory = _.find(state.history || [], h => h.type === 'drawer');

                        // Set the Onyx key `DEFAULT_DRAWER_STATUS`
                        // We only change the default drawer status `closed`` when
                        // 1. Current route is report screen and last default drawer status is open
                        // 2. When current route is report screen and we do refresh there`s no history saved.
                        //    So we check if last drawer status is `closed` we keep the drawer status to `closed`
                        // Other than that will be opening set drawer status to `open`
                        if ((hasDrawerHistory && state.default === 'open') || (!hasDrawerHistory && state.default === 'closed')) {
                            App.setDefaultDrawerStatus('closed');
                        } else {
                            App.setDefaultDrawerStatus('open');
                        }
                    },
                }}
```
4. Now we return the default drawer status to `Drawer.Navigator`. The drawer status value will be from Onyx key `DEFAULT_DRAWER_STATUS` and when url is open from home route we return `open` status.
https://github.com/Expensify/App/blob/dee673570654cd14aae5c36230aeda69bc04db13/src/libs/Navigation/Navigation.js#L77-L80
```js
    if (isSmallScreenWidth) {
        const path = getPathFromState(navigationRef.current.getState(), linkingConfig.config);
        return path === '/' ? 'open' : defaultDrawerStatus;
    }
```

The result will be https://github.com/Expensify/App/issues/8126#issuecomment-1104168082 and https://github.com/Expensify/App/issues/8126#issuecomment-1069452614

https://user-images.githubusercontent.com/25520267/166974869-4a4918ae-9699-408a-a37e-9d5a69ade4b0.mov


https://user-images.githubusercontent.com/25520267/166975033-1a9d00de-4093-401c-af2f-2296bbcbe83b.mov

**Comment by mollfpr** [comment]
@parasharrajat Any thought on this https://github.com/Expensify/App/issues/8126#issuecomment-1118839624 ?

**Comment by NicMendonca** [comment]
@parasharrajat bump ☝️

**Comment by parasharrajat** [comment]
Thanks for the proposal but this does not seem durable. I do not understand the changes in screenListeners. what are the reasons for using that code? ScreenListener runs on each screen which could be part of any navigator. This code will run on any navigator.

> // Set the Onyx key `DEFAULT_DRAWER_STATUS`
                        // We only change the default drawer status `closed`` when
                        // 1. Current route is report screen and last default drawer status is open
                        // 2. When current route is report screen and we do refresh there`s no history saved.
                        //    So we check if last drawer status is `closed` we keep the drawer status to `closed`
                        // Other than that will be opening set drawer status to `open
                        
 Why? for all conditions.                       

Also, I need to ask someone from team to review your proposal as well. I am not sure how suitable is this approach?

**Comment by mollfpr** [comment]
> Thanks for the proposal but this does not seem durable. I do not understand the changes in screenListeners. what are the reasons for using that code? ScreenListener runs on each screen which could be part of any navigator. This code will run on any navigator.
> 
> > // Set the Onyx key `DEFAULT_DRAWER_STATUS`
> > // We only change the default drawer status `` closed`` when                         // 1. Current route is report screen and last default drawer status is open                         // 2. When current route is report screen and we do refresh there ``s no history saved.
> > //    So we check if last drawer status is `closed` we keep the drawer status to `closed`
> > // Other than that will be opening set drawer status to `open
> 
> Why? for all conditions.
> 
> Also, I need to ask someone from team to review your proposal as well. I am not sure how suitable is this approach?

@parasharrajat  Yes, the choose is either set onyx key on screen listener or at the lib when try to change drawer state.

The reason why I use screenListener is to make it cleaner to handle set onyx value. If you prefer to set the value on drawer state change we can combine my other proposal to do that.

**Comment by mollfpr** [comment]
@parasharrajat We can do this instead to handle drawer state.

Set `ONYXKEYS.DEFAULT_DRAWER_STATUS` to `open` when the drawer is gonna be opened.
https://github.com/Expensify/App/blob/7919e24c808dea9dedfe4a2d513365cf092cc1e1/src/libs/Navigation/Navigation.js#L56-L63

Set `ONYXKEYS.DEFAULT_DRAWER_STATUS` to `closed` when navigating to the report route.
https://github.com/Expensify/App/blob/7919e24c808dea9dedfe4a2d513365cf092cc1e1/src/libs/Navigation/Navigation.js#L143-L146

**Comment by parasharrajat** [comment]
https://github.com/Expensify/App/issues/8126#issuecomment-1123095585 This looks better. You mean to do this instead of point 3. What else can be done for point 4? Did you check this with navigation? it should not break the navigation of the app.

**Comment by mollfpr** [comment]
@parasharrajat Yeah, we replace point 3 with this https://github.com/Expensify/App/issues/8126#issuecomment-1123095585 instead. 

Point 4 is to handle drawer state from URL. Currently, I only handle when the path is `/` and will return the drawer state to `open`. Other than that will return value from onyx which the last drawer states the user.

> Did you check this with navigation? it should not break the navigation of the app.

For now, nothing different happens with the navigation. My proposal does nothing to change the navigation behavior. The only thing my proposal does is save the drawer state to onyx and return it to drawer `defaultStatus`.

**Comment by thienlnam** [comment]
@parasharrajat any comments on the above?

**Comment by parasharrajat** [comment]
I will update you by tomorrow. I need to test a few things.

**Comment by parasharrajat** [comment]
I doubt that https://github.com/Expensify/App/issues/8126#issuecomment-1123095585 will always work. What if the drawer is opened/closed based on state change but from our action? will this be tracked in Onyx if we do https://github.com/Expensify/App/issues/8126#issuecomment-1123095585? 


If we follow your suggestion of a screen listener. Could you please explain that code to me? I don't see anything telling me in the comments why you add the next line of code.

How do you know that the current state is related to the drawer navigator? You haven't added anything in the code that only runs the code only for the drawer navigator. ScreenListeners runs on each navigator.

**Comment by mollfpr** [comment]
@parasharrajat Thank you for checking my proposal.

> I doubt that https://github.com/Expensify/App/issues/8126#issuecomment-1123095585 will always work. What if the drawer is opened/closed based on state change but from our action? will this be tracked in Onyx if we do https://github.com/Expensify/App/issues/8126#issuecomment-1123095585?

You might be right. We have custom action to reset the route accordingly.

> If we follow your suggestion of a screen listener. Could you please explain that code to me? I don't see anything telling me in the comments why you add the next line of code.

> How do you know that the current state is related to the drawer navigator? You haven't added anything in the code that only runs the code only for the drawer navigator. ScreenListeners runs on each navigator.

In my proposal, we use `screenListener` on `Drawer.Navigator` which means only listening to the screen changing on the drawer. I check it and it only calls `screenListener` when the drawer change either open/closed.

https://user-images.githubusercontent.com/25520267/168837030-714d83c3-b31f-4f99-a04b-f27c443cd7a2.mp4

I also think this https://github.com/Expensify/App/issues/8126#issuecomment-1118839624 is a more fit solution than my first proposal.

**Comment by mollfpr** [comment]
@NicMendonca applied!

**Comment by marcaaron** [comment]
Just to clarify here... the issue we're fixing is that if you refresh the page on a chat while on mobile web you land back on the sidebar? Does that not sound like an issue with the `react-navigation` library?

**Comment by parasharrajat** [comment]
This is a kind of missing feature from react-navigation. The way Drawer is built over Tab-navigator prevents it from creating browser history entries. We can't control its state via url or configuration.


IMO, The goto answer from react-navigation maintainer would be to use a custom router for drawer navigator to meet our needs. I also think that we should also create a custom router. It can also help us in getting away from custom actions. I previously experimented with it but it is quite complex.


But I agree that first we should create an issue on the upstream repo and get the suggestion of what type of solution, they are expecting. Are they willing to change it?

**Comment by marcaaron** [comment]
>The reason why we land back to LHN is because we return ‘open’ to the ‘defaultStatus’ of Drawer Navigator.

If we refreshing the page and navigating to a report why are we passing `'open'`. We should initialize it as `'closed'` in that case? Or am I thinking about this wrong?

>This is a kind of missing feature from react-navigation

Which feature is missing exactly?

>The way Drawer is built over Tab-navigator prevents it from creating browser history entries. We can't control its state via url or configuration.

Can you elaborate on this? I'm not sure why browser history entries are relevant. And also do not understand why we can't control the state of the drawer based on the url.

>The goto answer from react-navigation maintainer would be to use a custom router for drawer navigator to meet our needs. I also think that we should also create a custom router.

To clarify, are you saying that we presented this particular problem and the suggestion was to use a custom router and we took that feedback and decided to **not** do what the maintainer suggested to do?

**Comment by parasharrajat** [comment]
To correctly answer your questions, I will gather more information/facts from `react-navigation` repo which can be used to validate our analysis.

**Comment by thienlnam** [comment]
Any update here @parasharrajat regarding info from the react-navigation repo?

**Comment by mollfpr** [comment]
Nothing new from me, except for some improvement to the code. Honestly, I get confused about the expected result we looking for. 

We need more people to join the discussion.

Here’s the slack chat https://expensify.slack.com/archives/C01GTK53T8Q/p1653068209660299

**Comment by thienlnam** [comment]
Pinged the slack conversation, hoping to get some more input

**Comment by MTN718** [comment]
still available submit proposal?

**Comment by marcaaron** [comment]
@MTN718 not at the moment. But you can look for issues with the **Help Wanted** label.

**Comment by mollfpr** [comment]
## Recap
_**this issue is still reproducible**_

In our [slack conversation](https://expensify.slack.com/archives/C01GTK53T8Q/p1653068209660299), we agree to use the URL as a source of truth and we decide not to use the drawer tracker to make it simple.

The possible case will be handled:
- When the user opens the App from URL `/`, the LHN will be shown ✅
- When a user opens the App from the URL is `/r/reportID`, the report screen will be shown ❌
- When a user on the LHN (current URL `/r/reportID`) then refresh the page, the report screen will be shown ❌
- When a user is on the report screen (current URL /r/reports) and then refresh the page, the report screen will be shown ❌

_**✅ Is current behavior, ❌ Not yet implemented**_

For the next step, I will close the current PR (in case we need it in the future) and create a new one with the solution.
```
const path = getPathFromState(navigationRef.current.getState(), linkingConfig.config);
return path === '/' ? CONST.DRAWER_STATUS.OPEN : CONST.DRAWER_STATUS.CLOSED;
```
We will open the drawer when the user accesses the App from the root URL `/` and other than that we close the drawer to show the `ReportScreen`.

If you are all on board with this send this 🚀 or 👎 for other thoughts.

Thank you!

cc @marcaaron @thienlnam @parasharrajat

**Comment by marcaaron** [comment]
>When a user is on the report screen (current URL /r/reports) and then refresh the page, the report screen will be shown ❌

I don't know what `r/reports` is - not sure if we have a route like this. I think you just mean we are on a report and then refresh. Other than that, LGTM.

**Comment by mollfpr** [comment]
@marcaaron Yeah, it’s ‘/r/reportID’ 😅

**Comment by mollfpr** [comment]
@marcaaron @thienlnam @parasharrajat 

On the desktop minimize to mobile layout, when we refresh the app what should be expected? Is showing the LHN or Report Screen?

This is when `defaultStatus` drawer is set to `close`.
https://user-images.githubusercontent.com/25520267/174280490-6eeb574e-6f8a-47ff-a331-ba536984662a.mov

**Comment by parasharrajat** [comment]
It should behave as MWeb.

**Comment by thienlnam** [comment]
PR is being worked on

**Comment by adelekennedy** [comment]
not overdue! waiting for the payment ping

**Comment by thienlnam** [comment]
PR is still in review / revisions

**Comment by thienlnam** [comment]
Same as above, @mollfpr any update on the PR revisions?

**Comment by mollfpr** [comment]
@thienlnam I push a new commit and just waiting for a review.

**Comment by thienlnam** [comment]
cc @parasharrajat

**Comment by adelekennedy** [comment]
not overdue!

**Comment by thienlnam** [comment]
Pending review from @parasharrajat

**Comment by adelekennedy** [comment]
pending review

**Comment by thienlnam** [comment]
Been going through reviews, and now is pending another review from @parasharrajat 

Here's the PR link for reference https://github.com/Expensify/App/pull/9483

**Comment by adelekennedy** [comment]
What's the status on this, still pending review?

**Comment by adelekennedy** [comment]
Should we put this on HOLD for now? @parasharrajat @thienlnam

**Comment by adelekennedy** [comment]
@parasharrajat @thienlnam is there anything I should be watching for with this issue? Or can we close this out?

**Comment by thienlnam** [comment]
This is holding on high level changes with our navigation implementation - https://github.com/Expensify/App/issues/11768

However, we're unsure if this will remain an issue after those changes.  We've spent a lot of time on this proposal and solution and so let's keep this issue open (on a monthly), and see where we're at after https://github.com/Expensify/App/issues/11768.  But either way we should be compensating mollfpr for their time if we end up closing this or not

**Comment by JmillsExpensify** [comment]
Snagging this since I'm in the tracking issue.

**Comment by JmillsExpensify** [comment]
See the tracking issue for the latest.

**Comment by JmillsExpensify** [comment]
Still holding.

**Comment by JmillsExpensify** [comment]
Still held as this initiative goes through the design process.

**Comment by JmillsExpensify** [comment]
Still holding on the larger react navigation project.

**Comment by JmillsExpensify** [comment]
Still on hold for the larger navigation re-write.

**Comment by JmillsExpensify** [comment]
Same as above.

**Comment by JmillsExpensify** [comment]
Still on hold for navigation. No ETA.

**Comment by JmillsExpensify** [comment]
Same same

**Comment by thienlnam** [comment]
Same as above

**Comment by JmillsExpensify** [comment]
Same same.

**Comment by thienlnam** [comment]
The navigation PR has been merged - now we just need to see if this is still a bug and needs to be addressed

**Comment by phivh** [comment]
Sorry, do we consider bounty of reported this issue? 
@JmillsExpensify @thienlnam

**Comment by mountiny** [comment]
I think we should pay for reporting here

**Comment by JmillsExpensify** [comment]
Agree on that. This shouldn’t have been closed. I’m aiming to get through testing issues early this week, at which point I’ll issue payment.

**Comment by phivh** [comment]
> Agree on that. This shouldn’t have been closed. I’m aiming to get through testing issues early this week, at which point I’ll issue payment.

Gentle bump on this @JmillsExpensify

**Comment by JmillsExpensify** [comment]
@phivh Can you please re-test this issue, including video, and comment confirming whether the issue is still present? Thanks!

**Comment by JmillsExpensify** [comment]
Gentle bump on that last comment. Thank you!

**Comment by phivh** [comment]
> @phivh Mind testing this issue again per the above comment and we'll ideally be able to close this issue out and issue payment?

Sorry. I was occupied on something, will find time to test this week.

**Comment by JmillsExpensify** [comment]
Great thank you!

**Comment by JmillsExpensify** [comment]
Waiting on test

**Comment by JmillsExpensify** [comment]
Same

**Comment by mountiny** [comment]
I think this is not repro anymore and we should close this out, i dont know if we need to pay anything out here

**Comment by mallenexpensify** [comment]
Wow, this bug is over two years old and @mollfpr was hired close to two years ago 🤯 
I'm going to close as it's not reproducible.  @parasharrajat , @mollfpr , if you believe you're due compensation, can you please propose the amount, based on 25, 50 or 100% of the job price? Please provide reasoning too (ie. link the PR if one was created).   We'll reopen this if needed.

**Comment by mollfpr** [comment]
I don't remember much of the detail but considering the length of the discussion on the proposal and the PR and how long the issue is held, 75% would be great! This was the highest job I ever got, so I would expect that I could finish the job at that time.

**Comment by mallenexpensify** [comment]
Lot to unpack here.  @JmillsExpensify , tagging you since  you're the BZ and might have more context. 

It looks like
- The job got raised to [$8k at one point](https://github.com/Expensify/App/issues/8126#event-6536239107) and @mollfpr was hired and @parasharrajat to review. @mollfpr worked on two PRs ([one](https://github.com/Expensify/App/pull/9483) and [two](https://github.com/Expensify/App/pull/9076))
- We then [put the issue on hold](https://github.com/Expensify/App/issues/8126#issuecomment-1298928328) pending [this issue](https://github.com/Expensify/App/issues/11768). 
- Issue was closed because it wasn't reproducible. 
- We should pay something and we need to figure out what the fair amount is. 

I'm leaning towards 50% for each @mollfpr and @parasharrajat .  It's quite a lot of money, $4k each, but we were increasing prices frequently a year+ ago and I don't think either should be penalized because this one stayed open for so long (and we're paying less now).  @JmillsExpensify , @thienlnam, what do you think? 
-

**Comment by JmillsExpensify** [comment]
I'm fine with @mallenexpensify's summary, though Matt can you please type out the payment summary. I can't approve my own summaries and both request in Expensify.

**Comment by mallenexpensify** [comment]
Contributor: @mollfpr due $4000 via NewDot
Contributor+: @parasharrajat due $4000 via NewDot

Thanks!

**Comment by JmillsExpensify** [comment]
$4,000 approved for @parasharrajat
